homeassistant/components/systemmonitor/sensor.py

```diff
@@ -334,39 +340,84 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the system monitor sensors."""
+    processes = [
+        resource[CONF_ARG]
+        for resource in config[CONF_RESOURCES]
+        if resource[CONF_TYPE] == "process"
+    ]
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data={"processes": processes}
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up System Montor sensors based on a config entry."""
     entities = []
     sensor_registry: dict[tuple[str, str], SensorData] = {}
+    for _type, sensor_description in SENSOR_TYPES.items():
+        if _type.startswith("disk_"):
+            arguments = await hass.async_add_executor_job(get_all_disk_mounts)
```

Comment:
We don't need to get the disk mounts more than once. Same for network interfaces and cpu temperature. Break out those calls from the loop.